freeipa-client/src/main/java/com/sequenceiq/freeipa/client/FreeIpaClient.java

```diff
@@ -207,14 +208,21 @@ public void updateUserPasswordMaxExpiration(String user) throws FreeIpaClientExc
         updateUserPasswordExpiration(user, Optional.empty());
     }
 
-    public User userSetPasswordHash(String user, String hashedPassword,
-            String unencryptedKrbPrincipalKey, Optional<Instant> expiration) throws FreeIpaClientException {
-        String passwordExpirationDate = formatDate(expiration);
-        Map<String, Object> params =
-                Map.of("setattr", List.of(
-                        "cdpHashedPassword=" + hashedPassword,
-                        "cdpUnencryptedKrbPrincipalKey=" + unencryptedKrbPrincipalKey,
-                        "krbPasswordExpiration=" + passwordExpirationDate));
+    public User userSetWorkloadCredentials(String user, String hashedPassword,
+            String unencryptedKrbPrincipalKey, Optional<Instant> expiration,
+            List<String> sshPublicKeys) throws FreeIpaClientException {
+        Map<String, Object> params = new HashMap<>();
+        List<String> attributes = new ArrayList<>();
+
+        if (null != hashedPassword && !hashedPassword.isBlank()) {
```

Comment:
apache StringUtils.isNotBlank(hashPassword) is a shorter

Reply:
thanks!